### DISCUSSION TITLE
How do I matchFulfilled/matchRejected on all queries from an API?

### DISCUSSION THREAD

**Discussion by arp** [reduxjs/redux-toolkit#4734] [Q&A] [answered]
I sometimes do this in `extraReducers`:
```
builder.addMatcher(backendApi.endpoints.getData.matchFulfilled, (state, action) => { ... })
```

But, how do I do the same to match fulfilled/rejected request for all endpoints in an API.

Something like `backendApi.matchFulfilled` (which doesn't work, I tried :P )

**Comment by markerikson** [comment, marked as answer]
You'd need to write a custom matcher that checks for, roughly, `action.type.startsWith("myApi/") && action.type.endsWith("/fulfilled")`.
